src/canopy/llm/cohere.py

```diff
@@ -227,25 +248,37 @@ def _convert_model_params(self, openai_model_params: dict) -> dict:
         common_params = [
             "model",
             "frequency_penalty",
+            "logit_bias",
             "max_tokens",
             "presence_penalty",
             "stream",
             "temperature",
         ]
+
+        equivalent_params = {
+            "top_p": "p",
+            "user": "user_name",
+        }
+
         converted_model_params = {}
 
         for param in common_params:
             if param in openai_model_params:
                 converted_model_params[param] = openai_model_params.pop(param)
 
+        for param, equivalent in equivalent_params.items():
+            if param in openai_model_params:
+                converted_model_params[equivalent] = openai_model_params.pop(param)
```

Comment:
@igiloh-pinecone The problem with the suggested change is that it's not allowed to pop items from a dict while looping over it. Even with `for param in openai_model_params.keys():` it's still a runtime error. That is why I have slightly modified the suggested change to cast it to a list, so that we are iterating over the list, not the dict. It's effectively the same, though.

Reply:
Yeah, you're absolutely right, I missed that.